tests/nat/observability/exporter/test_span_exporter.py

```
@@ -573,3 +575,177 @@ async def test_usage_info_without_token_usage(self, span_exporter):
+
+        result = exporter._to_dict(CustomObj())
+        assert result == "custom_string_repr"
+        assert isinstance(result, str)
+
+    def test_exception_during_serialization_falls_back_to_str(self, exporter):
+        """If JSON serialization fails, falls back to str()."""
+
+        class BrokenModel:
+            """Object with model_dump that returns non-serializable data."""
+
+            def model_dump(self, **kwargs):
```

Ruff (0.14.14)

Unused method argument: `kwargs`

(ARG002)